temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java

```diff
@@ -70,18 +70,21 @@
 
   private final Thread.UncaughtExceptionHandler uncaughtExceptionHandler =
       (t, e) -> {
-        if (e instanceof StatusRuntimeException) {
-          StatusRuntimeException te = (StatusRuntimeException) e;
-          if (te.getStatus().getCode() == Status.Code.DEADLINE_EXCEEDED) {
-            log.warn("Failure in thread " + t.getName(), e);
-            return;
-          }
-          if (te.getCause() instanceof InterruptedException) {
-            log.debug("Failure in thread " + t.getName(), e);
-            return;
+        if (!pollExecutor.isTerminating() || !shouldIgnoreDuringTermination(e)) {
+          if (e instanceof StatusRuntimeException) {
+            StatusRuntimeException te = (StatusRuntimeException) e;
+            if (te.getStatus().getCode() == Status.Code.DEADLINE_EXCEEDED) {
+              log.warn("Failure in thread {}", t.getName(), e);
+              return;
+            }
           }
+          log.error("Failure in thread {}", t.getName(), e);
```

Comment:
Might be nice to extract this to a `logPollErrors` function or something, to make it clear that the outermost `if` is all about guarding logging stuff.